frontend/src/types.ts

```diff
@@ -605,38 +605,68 @@ export enum ProgressionStep {
   SERVER_REQUESTED = 'SERVER_REQUESTED',
   POD_CREATED = 'POD_CREATED',
   POD_ASSIGNED = 'POD_ASSIGNED',
+  POD_PROBLEM = 'POD_PROBLEM',
   PVC_ATTACHED = 'PVC_ATTACHED',
   INTERFACE_ADDED = 'INTERFACE_ADDED',
   PULLING_NOTEBOOK_IMAGE = 'PULLING_NOTEBOOK_IMAGE',
   NOTEBOOK_IMAGE_PULLED = 'NOTEBOOK_IMAGE_PULLED',
   NOTEBOOK_CONTAINER_CREATED = 'NOTEBOOK_CONTAINER_CREATED',
+  NOTEBOOK_PROBLEM = 'NOTEBOOK_PROBLEM',
   NOTEBOOK_CONTAINER_STARTED = 'NOTEBOOK_CONTAINER_STARTED',
   PULLING_OAUTH = 'PULLING_OAUTH',
   OAUTH_PULLED = 'OAUTH_PULLED',
   OAUTH_CONTAINER_CREATED = 'OAUTH_CONTAINER_CREATED',
+  OAUTH_PROBLEM = 'OAUTH_PROBLEM',
   OAUTH_CONTAINER_STARTED = 'OAUTH_CONTAINER_STARTED',
   SERVER_STARTED = 'SERVER_STARTED',
 }
 
-export const ProgressionStepTitles = {
+export const ProgressionStepTitles: Record<ProgressionStep, string> = {
   [ProgressionStep.SERVER_REQUESTED]: 'Server requested',
   [ProgressionStep.POD_CREATED]: 'Pod created',
   [ProgressionStep.POD_ASSIGNED]: 'Pod assigned',
+  [ProgressionStep.POD_PROBLEM]: 'There was a problem with the pod',
   [ProgressionStep.PVC_ATTACHED]: 'PVC attached',
   [ProgressionStep.INTERFACE_ADDED]: 'Interface added',
   [ProgressionStep.PULLING_NOTEBOOK_IMAGE]: 'Pulling workbench image',
   [ProgressionStep.NOTEBOOK_IMAGE_PULLED]: 'Workbench image pulled',
   [ProgressionStep.NOTEBOOK_CONTAINER_CREATED]: 'Workbench container created',
+  [ProgressionStep.NOTEBOOK_PROBLEM]: 'There was a problem with the notebook',
   [ProgressionStep.NOTEBOOK_CONTAINER_STARTED]: 'Workbench container started',
   [ProgressionStep.PULLING_OAUTH]: 'Pulling oauth proxy',
   [ProgressionStep.OAUTH_PULLED]: 'Oauth proxy pulled',
   [ProgressionStep.OAUTH_CONTAINER_CREATED]: 'Oauth proxy container created',
+  [ProgressionStep.OAUTH_PROBLEM]: 'There was a problem with Oauth',
   [ProgressionStep.OAUTH_CONTAINER_STARTED]: 'Oauth proxy container started',
   [ProgressionStep.SERVER_STARTED]: 'Server started',
 };
 
+export const AssociatedSteps: Record<string, ProgressionStep[]> = {
```

Comment:
```suggestion
export const AssociatedSteps: Record<Partial<ProgressionStep>, ProgressionStep[]> = {
```
Does this work?

Reply:
Yeah I fought with that for a bit so we could either stick with 'string' or I could do something like this (it didn't enjoy the Partial):
`export type AssociatedStepKeys =
  | ProgressionStep.NOTEBOOK_CONTAINER_STARTED
  | ProgressionStep.OAUTH_CONTAINER_STARTED
  | ProgressionStep.POD_ASSIGNED
  | ProgressionStep.SERVER_STARTED;

export const AssociatedSteps: Record<AssociatedStepKeys, ProgressionStep[]> = {`
But then we'd need to update the AssociatedStepKeys if there are any changes there but it also might be better than just plain 'string'

Reply:
Hmmm... okay, I'll look -- I'm not sure we want to tie the type to the specific use-case, more define the structure... TS sometimes has issues with this... lemme see what I can do when I check this out.

Reply:
```suggestion
export const AssociatedSteps: { [key in ProgressionStep]?: ProgressionStep[] } = {
```
Record is such a useful way to write this block -- but effectively it seems it cannot handle partial keys -- looks TS cares more about "not which keys" but "which keys will have values" -- so you have to use the optional `?` indicator to set this up -- this is a slight step forward so that `foo` key cannot be one in this map -- as it requires it to be part of the Enum -- so if we ever remove an Enum value, TS will catch that we are still using it here -- which might happen anyways, as we key off the enum -- but at least it will limit it to only values that will "ever work" and we don't have to assume all `string`s are valid.